demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,874 @@
+{
+    FILE * file;
+    size_t length = 0;
+    bool status = true;
+
+    /* Get the file descriptor for the CSR file. */
+    file = fopen( PROVISIONING_CSR_PATH, "rb" );
+
+    if( file == NULL )
+    {
+        LogError( ( "Error opening file at PROVISIONING_CSR_PATH: %s. Error: %s.",
+                    PROVISIONING_CSR_PATH, strerror( errno ) ) );
+        status = false;
+    }
+    else
+    {
+        int result;
+        /* Seek to the end of the file, so that we can get the file size. */
+        result = fseek( file, 0L, SEEK_END );
+
+        if( result == -1 )
+        {
+            LogError( ( "Failed while moving to end of the certificate signing request file. Path: %s. Error: %s.",
+                        PROVISIONING_CSR_PATH, strerror( errno ) ) );
+            status = false;
+        }
+        else
+        {
+            long lenResult = -1;
+            /* Get the current position which is the file size. */
+            lenResult = ftell( file );
+
+            if( lenResult == -1 )
+            {
+                LogError( ( "Failed to get length of certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+            else
+            {
+                length = ( size_t ) lenResult;
+            }
+        }
+
+        if( status == true )
+        {
+            if( length > bufferLength )
+            {
+                LogError( ( "Buffer too small for certificate signing request. Buffer size: %ld. Required size: %ld.",
+                            bufferLength, length ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            /* Return to the beginning of the file. */
+            result = fseek( file, 0L, SEEK_SET );
+
+            if( result == -1 )
+            {
+                LogError( ( "Failed to move to beginning of certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            size_t written = 0;
+            /* Read the CSR into our buffer. */
+            written = fread( pBuffer, 1, length, file );
+
+            if( written != length )
+            {
+                LogError( ( "Failed reading certificate signing request file. Path: %s. Error: %s.",
+                            PROVISIONING_CSR_PATH, strerror( errno ) ) );
+                status = false;
+            }
+            else
+            {
+                *pOutCsrLength = length;
+            }
+        }
+
+        fclose( file );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool waitForResponse( void )
+{
+    bool status = false;
+
+    responseStatus = ResponseNotReceived;
```

Comment:
It would be good to explicitly mention that if a response is received, then the publish callback would update the status variable. 
Without the comment, it may not be obvious to a reader about how the `responseStatus` variable is getting updated.